### PR TITLE
Check for duplicate member tag names in serialization generation

### LINKED ISSUE #8381: json_deserialize_sql could not round-trip parsed results from json_serialize_sql
### What happens?

json_deserialize_sql seems could not handle certain parsed node types 
==
```
v0.8.2-dev2133 726b6d1566
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D  select json_deserialize_sql(json_serialize_sql('
> SELECT
>     DATE_TRUNC(''month'', p.time) AS cohort_month,
>     COUNT(DISTINCT p.company_id) AS active_users,
>     SUM(p.amount) AS total_purchase_amount
> FROM
>     Purchase p
> JOIN
>     (SELECT
>         user_id,
>         MIN(time) AS first_purchase_time
>      FROM
>         Purchase
>      GROUP BY
>         user_id) fp
> ON
>     p.user_id = fp.user_id
>     AND DATE_TRUNC(''month'', p.time) = DATE_TRUNC(''month'', fp.first_purchase_time)
> GROUP BY
>     DATE_TRUNC(''month'', p.time)
> ORDER BY
>     cohort_month;
> '));
Error: Not implemented Error: Enum value: 'JOIN' not implemented
```

### To Reproduce
1. Start Duckdb CLI
2. Cut-n-paste 
```
 select json_deserialize_sql(json_serialize_sql('
SELECT
    DATE_TRUNC(''month'', p.time) AS cohort_month,
    COUNT(DISTINCT p.company_id) AS active_users,
    SUM(p.amount) AS total_purchase_amount
FROM
    Purchase p
JOIN
    (SELECT
        user_id,
        MIN(time) AS first_purchase_time
     FROM
        Purchase
     GROUP BY
        user_id) fp
ON
    p.user_id = fp.user_id
    AND DATE_TRUNC(''month'', p.time) = DATE_TRUNC(''month'', fp.first_purchase_time)
GROUP BY
    DATE_TRUNC(''month'', p.time)
ORDER BY
    cohort_month;
'));
```

### OS:

Mac OS

### DuckDB Version:

v0.8.2-dev2133 726b6d1566 (private build from source)

### DuckDB Client:

DuckDB CLI

### Full Name:

Min Wei

### Affiliation:

None

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by Maxxen**
Also adjust some tag names to no longer conflict within their inheritance chain.

Closes #8381

### CODE REVIEW

File: scripts/generate_serialization.py [hunk 1]
```diff
@@ -162,6 +162,14 @@ def generate_return(class_entry):
         return '\treturn std::move(result);'
 
 
+def detect_duplicate_member_names(class_entry):
```
Comment (Mause) [inline scripts/generate_serialization.py:165]: This function doesn't do anything?

File: scripts/generate_serialization.py [hunk 2]
```diff
@@ -529,6 +537,28 @@ def generate_class_code(class_entry):
                     raise Exception(f"Duplicate enum entry \"{enum_entry}\"")
                 base_class_object.children[enum_entry] = new_class
 
+    # Ensure that there are no duplicate names in the inheritance tree
+    for base_class in base_classes:
+        if base_class.base is None:
+            # Root base class, now traverse the children
+            def traverse_children(node, parents: list, seen: set):
```
Comment (Mause) [inline scripts/generate_serialization.py:544]: Move this function out of the loop so it doesn't get defined (again) on each iteration?

**Review by Mause — CHANGES_REQUESTED** [review]

File: scripts/generate_serialization.py (hunk 1 above)
Comment (Maxxen) [reply scripts/generate_serialization.py:165]: Ah crap, was going to remove this

**Comment by Mytherin** [comment]
Thanks! Looks great

**Pull request merged**